exareme2/worker/flower/starter/flower_service.py

```diff
@@ -1,53 +1,48 @@
-from exareme2 import ALGORITHM_FOLDERS
 from exareme2.algorithms.flower.process_manager import FlowerProcess
 from exareme2.worker import config as worker_config
 from exareme2.worker.utils.logger import get_logger
 from exareme2.worker.utils.logger import initialise_logger
 
-# Dictionary to keep track of running processes
-running_processes = {}
-SERVER_ADDRESS = "0.0.0.0:8080"
+SERVER_PORT = "8080"
 
 
 @initialise_logger
-def start_flower_client(request_id: str, algorithm_name, worker_id) -> int:
+def start_flower_client(request_id: str, algorithm_name, server_ip) -> int:
     env_vars = {
         "MONETDB_IP": worker_config.monetdb.ip,
         "MONETDB_PORT": worker_config.monetdb.port,
         "MONETDB_USERNAME": worker_config.monetdb.local_username,
         "MONETDB_PASSWORD": worker_config.monetdb.local_password,
         "MONETDB_DB": worker_config.monetdb.database,
-        "SERVER_ADDRESS": SERVER_ADDRESS,
+        "SERVER_ADDRESS": f"{server_ip}:{SERVER_PORT}",
```

Comment:
Why are we passing the `server_ip`  as an argument and the `server_port` as an env variable?

I would expect both to follow the same path, in the future as well we will need to pass the port from the controller as well. Maybe pass both from the controller?